### DISCUSSION TITLE
Integrate Alfa with Selenium

### DISCUSSION THREAD

**Discussion by smithapillai** [Siteimprove/alfa#310] [Questions] [answered]
Hello Team,
I am looking to integrate Alfa into my existing Selenium based framework. Any pointers for this? I dont seem find a package for selenium integration. 
Any help will be appreciated !

**Comment by kasperisager** [comment, marked as answer]
Hi there and thanks for reaching out! We don't yet offer an integration for the JavaScript-based Selenium bindings, though it's definitely something I'd be interested in investigating. The closest we currently have is an integration for WebdriverIO, which is capable of connecting to a Selenium grid as far as I can tell. Are you interacting with Selenium through the JavaScript bindings?

**Comment by smithapillai** [comment]
Thanks for the reply !! Can you please let me know the details about the Webdriver integration. I will check that if that can be embedded in our framework.
Thanks Again !

**Comment by kasperisager** [comment]
No problem! You can see an example of the integration in use here: https://github.com/Siteimprove/alfa/tree/master/docs/examples/page-testing/webdriver

**Comment by smithapillai** [comment]
I was expecting an alfa library that can be added to pom file and my java code can make use of the alfa methods. I think thats not possible right?

**Comment by kasperisager** [comment]
Ah, I'm afraid not; for now, we will only provide integrations for JavaScript as Alfa itself is written in JavaScript. One possibility would however be to create a script that runs an audit using Alfa, bundle that script using a tool like [Rollup](https://rollupjs.org/guide/en/), and then load that bundle into the browser from Selenium.
